### PR TITLE
feat(eos-verifier): Add native eos verifier

### PR DESCRIPTION
**Pull request by Dimo99**

### CODE REVIEW

File: contracts/eos/verifier/src/cpp/nativeverifier/nativeverifier.cpp [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+#include <groth16/rapiduint256.hpp>
+
+using namespace eosio;
+
+static constexpr uint32_t ROOT_LENGTH = 32;
+static constexpr uint32_t PROOF_LENGTH = 384;
+
+static const char hex_chars[16] = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f'};
+
+class [[eosio::contract("nativeverifier")]] nativeverifier : public eosio::contract
```
Comment (EmilIvanichkovv) [inline contracts/eos/verifier/src/cpp/nativeverifier/nativeverifier.cpp:13]:
```suggestion
class [[eosio::contract("verifiernative")]] verifiernative : public eosio::contract
```
`verifiernative` seems like a better name?

File: contracts/eos/verifier/src/cpp/nativeverifier/nativeverifier.cpp [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+
+        // std::array<uint8_t, VERIFICATION_KEY_LENGTH> _vk;
+        // std::array<uint8_t, PROOF_LENGTH> _prf;
+        // std::array<uint8_t, ROOT_LENGTH> _current_header_root;
+        // std::array<uint8_t, ROOT_LENGTH> _new_optimistic_header_root;
+        // std::array<uint8_t, ROOT_LENGTH> _new_finalized_header_root;
+        // std::array<uint8_t, ROOT_LENGTH> _new_execution_state_root;
+        // std::array<uint8_t, 8> _new_slot;
+        // std::array<uint8_t, ROOT_LENGTH> _domain;
+
```
Comment (EmilIvanichkovv) [inline contracts/eos/verifier/src/cpp/nativeverifier/nativeverifier.cpp:99]: We don't need those lines?

File: contracts/eos/verifier/src/cpp/nativeverifier/nativeverifier.cpp
Comment (EmilIvanichkovv) [inline contracts/eos/verifier/src/cpp/nativeverifier/nativeverifier.cpp:1]: Double check and remove all debugging lines please

File: contracts/eos/verifier/src/cpp/nativeverifier/nativeverifier.cpp [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+                             {
+      std::array<uint8_t, 32> _current_header_root;
+
+      for (int pos = 0; pos < 32; pos++) {
+        std::copy(row.new_optimistic_header_roots[pos].begin(),
+                  row.new_optimistic_header_roots[pos].end(),
+                  _current_header_root.begin());
+        printhelper(_current_header_root);
+      } });
+    }
```
[the remark this answers is not in the document]
Comment (zah) [reply contracts/eos/verifier/src/cpp/nativeverifier/nativeverifier.cpp]: If these functions are useful for debugging, one option is to to enable them only in local builds through conditional compilation:

https://learn.microsoft.com/en-us/cpp/preprocessor/hash-if-hash-elif-hash-else-and-hash-endif-directives-c-cpp?view=msvc-170

[the remark this answers is not in the document]
Comment (EmilIvanichkovv) [reply contracts/eos/verifier/src/cpp/nativeverifier/nativeverifier.cpp]: We can query the contract using `cleos get table` and the result is more verbose and well structured, so my believe is we don't need need them.

**Pull request merged**